File: online-clothing-store-backend/src/main/java/com/app/service/ProductImageService.java
package com.app.service;

import java.io.IOException;
import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.app.dtos.ProductImageDTO;
import com.app.pojos.ProductImage;

public interface ProductImageService {
	
	public String addProductImage(ProductImageDTO productImageDTO) throws IOException;
	public byte[] getProductImage(Integer product,Integer ImageId);
	public List<byte[]> getAllProductImages(Integer productId);
	public String updateProductImageDetails(Integer productImageId, ProductImageDTO productImageDto) throws IOException;
	public String deleteProductImage(Integer productImageId);
	//public String uploadImage(Integer productImageId, MultipartFile image);
}

File: online-clothing-store-frontend/src/service/ProductImageService.js
import axios from 'axios';
class ProductImageService{
    baseurl="http://localhost:8080";

    

    getAllProductImages(){
        return axios.get(this.baseurl+"/product_images/product_id/3/image_id/1")
    }

    deleteProductImage(id){
        return axios.delete(this.baseurl+"ProductImage/"+id+".json")
    }
    addProductImage(productImage){
        return axios.post(this.baseurl+"ProductImage.json",productImage,
        {headers:{'content-type':'application/json'}})
    }
    updateProductImage(productImage){
        console.log("in service update",productImage);

        return axios.put(this.baseurl+"ProductImage/"+productImage.id+".json",productImage,
        {headers:{'content-type':'application/json'}})
    }

}

export default new ProductImageService();